online-server/src/main/java/eu/itesla_project/online/server/OnlineApplicationBean.java

```diff
@@ -57,285 +57,220 @@
+        OnlineWorkflowInfo wfi = workflows.get(workflowId);
+        return wfi.getSecurityRulesIndexesApplication();
+    }
+
+    public WcaContingenciesSynthesis getWcaContingencies(String workflowId) {
+        OnlineWorkflowInfo wfi = workflows.get(workflowId);
+        return wfi.getWcaContingencies();
+    }
+
+    public int getAvailableCores() {
+        int cores = 0;
+        if (remoteConnected && application != null)
+            cores = application.getAvailableCores();
+        return cores;
+    }
+
+    public WorkflowStatusEnum getWorkflowStatus(String workflowId) {
+        return workflows.get(workflowId).getStatus();
+    }
+
+    public Collection<String> getWorkFlowIds() {
+        return workflows.keySet();
+    }
+
+    public HashMap<String, OnlineWorkflowInfo> getWorkFlows() {
+        return workflows;
+    }
+
+    public boolean isJmxConnected() {
+        return (remoteConnected && application != null);
     }
 
-   
-
-	public boolean isWcaRunning(String workflowId) {
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.isWcaRunning();
-	}
-	
-	public HashMap<Integer, WorkStatus> getWorkStatus(String workflowId) {
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.getWorkStatus();
-	}
-	
-	public ContingencyStatesActionsSynthesis getStatesActions(String workflowId){
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.getStatesActions();
-	}
-	
-	public ContingencyStatesIndexesSynthesis getUnsafeContingencies(String workflowId){
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.getUnsafeContingencies();
-	}
-	
-
-	public IndexSecurityRulesResultsSynthesis getSecurityRulesIndexesApplication(String workflowId){
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.getSecurityRulesIndexesApplication();
-	}
-	
-	/*
-	
-	public String[] getStableContingencies(String workflowId){
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.getStableContingencies();
-	}
-
-	public UnstableContingenciesSynthesis getUnstableContingencies(String workflowId){
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.getUnstableContingencies();
-	}
-	*/
-	
-	public WcaContingenciesSynthesis getWcaContingencies(String workflowId){
-		OnlineWorkflowInfo wfi = workflows.get(workflowId);
-		return wfi.getWcaContingencies();
-	}
-
-	public int getAvailableCores() {
-		int cores=0;
-		if(remoteConnected && application !=null)
-			cores=application.getAvailableCores();
-		return cores;
-	}
-
-
-	public int getWorkflowStatus(String workflowId) {
-		return workflows.get(workflowId).getStatus();
-		
-	}
-	
-	
-	public Collection<String> getWorkFlowIds(){
-		return workflows.keySet();	
-	}
-	
-	
-	public HashMap<String, OnlineWorkflowInfo> getWorkFlows()
-	{
-		return workflows;		
-		
-	}
-	
-	public boolean isJmxConnected(){
-		return (remoteConnected && application !=null );
-	}
-	
-	
```

Comment:
A lot of input parameters could be null and cause NullPointerException. It could be interested to add assertion to test where it's needed